Develop

jmcur opened the pull request:
Fixed Inbox related methods.
Inbox methods dont use dummy data anymore.
Modify some variable names to match Android's.
Updated documentation.

Commit:
Modified parameter names

Commit:
Fixed Inbox related methods.
Inbox methods dont use dummy data anymore.
Modify some variable names to match Android's.
Updated documentation.

erikpoort commented on README.md:
```diff
@@ -82,4 +82,37 @@ Accengage.trackLead(label, value);
+
+/**
+* Get Inbox Messages
+* Returns an array of Messages.
```
Could you add the format of the message here or somewhere else where it makes sense?

erikpoort commented on README.md:
```diff
@@ -82,4 +82,37 @@ Accengage.trackLead(label, value);
+
+/**
+* Get Message
+* Returns a single message given an index.
```
Could you add the format of the message here or somewhere else where it makes sense?

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -25,17 +25,17 @@ @implementation RNAccengageModule
 #pragma mark - Permissions
 
 RCT_EXPORT_METHOD(
-                  hasPermissions:(RCTResponseSenderBlock)callback
```
This is a callback, not a promise:
https://facebook.github.io/react-native/docs/native-modules-ios.html#callbacks
https://facebook.github.io/react-native/docs/native-modules-ios.html#promises

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -181,66 +182,84 @@ - (void)getMessagesFromIndex:(int)pageIndex limit:(int)limit messageListCallback
+    if(_inbox == nil)
+    {
+        NSString *errorMessage = [NSString stringWithFormat:@"Inbox was null"];
+        NSString *operationCode = [NSString stringWithFormat:@"%@",@(AccengageCallIsLoading)];
```
Do you mean to use this operationCode?

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -181,66 +182,84 @@ - (void)getMessagesFromIndex:(int)pageIndex limit:(int)limit messageListCallback
+            }
+
+            //remove number of loaded messages when the service call had failed
+            [_loadedMessages setObject:@"" atIndexedSubscript:requestedIndex]; //change @"" to String
```
Interesting how Accengage iOS doesn't send an error message here

erikpoort reviewed (changes requested):
I have added some comments for you to look at :)
Mostly seems good 👍 

Can you double check the errors? I use 4 myself. Most errors is a safety check that should never happen, but if it happens, doesn't need custom errorhandling.

Could you also add those errors to the readme? So the user of the lib knows what to expect :)

Commit:
Added helper method for error messages.
Fix hasPermissions return attribute from promise to callback
Fix operationCode error on getMessagesFromIndex
README updated
